x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlDailyMaintenanceService.java

```diff
@@ -113,17 +122,34 @@ private synchronized void scheduleNext() {
     }
 
     private void triggerTasks() {
-        LOGGER.info("triggering scheduled [ML] maintenance tasks");
-        executeAsyncWithOrigin(client, ML_ORIGIN, DeleteExpiredDataAction.INSTANCE, new DeleteExpiredDataAction.Request(),
+        try {
+            LOGGER.info("triggering scheduled [ML] maintenance tasks");
+            executeAsyncWithOrigin(client, ML_ORIGIN, DeleteExpiredDataAction.INSTANCE, new DeleteExpiredDataAction.Request(),
                 ActionListener.wrap(
-                        response -> {
-                            if (response.isDeleted()) {
-                                LOGGER.info("Successfully completed [ML] maintenance tasks");
-                            } else {
-                                LOGGER.info("Halting [ML] maintenance tasks before completion as elapsed time is too great");
-                            }
-                        },
-                        e -> LOGGER.error("An error occurred during maintenance tasks execution", e)));
-        scheduleNext();
+                    response -> {
+                        if (response.isDeleted()) {
+                            LOGGER.info("Successfully completed [ML] maintenance tasks");
+                        } else {
+                            LOGGER.info("Halting [ML] maintenance tasks before completion as elapsed time is too great");
+                        }
+                    },
+                    e -> LOGGER.error("An error occurred during maintenance tasks execution", e)));
+            auditUnassignedMlTasks(clusterService.state());
+        } finally {
+            scheduleNext();
+        }
+    }
+
+    /**
+     * The idea of this is that if tasks are unassigned for days on end then they'll get a duplicate
+     * audit warning every day, and that will mean they'll permanently have a yellow triangle next
+     * to their entries in the UI jobs list.  (This functionality may need revisiting if the condition
+     * for displaying a yellow triangle in the UI jobs list changes.)
+     */
+    private void auditUnassignedMlTasks(ClusterState state) {
+        PersistentTasksCustomMetaData tasks = state.getMetaData().custom(PersistentTasksCustomMetaData.TYPE);
+        if (tasks != null) {
+            mlAssignmentNotifier.auditMlTasks(state.nodes(), tasks, tasks, true);
```

Comment:
It's a bit odd that we're passing the same `tasks` as both the `previousTasks` and the `CurrentTasks` here. Assuming this works properly, should we encapsulate that in `MlAssignmentNotifier` by making the current `auditMlTasks` method `private` and having another `public` one that only accepts a single `tasks` arg?